test/src/sdk-harness/test_projects/auth/mod.rs

```diff
@@ -94,3 +99,109 @@ async fn get_contracts() -> (
         wallet.lock(),
     )
 }
+
+#[tokio::test]
+async fn can_get_predicate_id() {
+    // Setup Wallets
+    let asset_id = AssetId::default();
+    let wallets_config = WalletsConfig::new_multiple_assets(
+        2,
+        vec![AssetConfig {
+            id: asset_id,
+            num_coins: 1,
+            coin_amount: 1_000,
+        }],
+    );
+    let wallets = &launch_custom_provider_and_get_wallets(wallets_config, None, None).await.unwrap();
+    let first_wallet = &wallets[0];
+    let second_wallet = &wallets[1];
+
+    // Setup Predciate
+    let hex_predicate_address: &str = "0x935a191561e0e9388c1b5dfc4626f6ecd98b2b3dd416a9a9e9ce2f0bb214f4b8";
+    let predicate_address = Address::from_str(hex_predicate_address).expect("failed to create Address from string");
+    let predicate_bech32_address = Bech32Address::from(predicate_address);
+    let predicate_data = AuthPredicateEncoder::encode_data(predicate_bech32_address);
+    let predicate: Predicate = Predicate::load_from("test_artifacts/auth_predicate/out/debug/auth_predicate.bin").unwrap()
+        .with_provider(first_wallet.try_provider().unwrap().clone())
+        .with_data(predicate_data);
+
+    // Next, we lock some assets in this predicate using the first wallet:
```

Comment:
We can simplify this to check for the truth value, no transfers needed. Simple tests are better